internalv2compat/compat.go

```diff
@@ -93,13 +99,27 @@ func V2TracingEnabled() bool {
 func SetV2TracingThriftClientMiddleware(middleware thrift.ClientMiddleware) {
 	v2Tracing.Lock()
 	defer v2Tracing.Unlock()
-	v2Tracing.thriftClientMiddleware = middleware
+	v2Tracing.thriftClientMiddlewareProvider = func(args ClientTraceMiddlewareArgs) thrift.ClientMiddleware {
+		return middleware
+	}
 }
 
 func V2TracingThriftClientMiddleware() thrift.ClientMiddleware {
 	v2Tracing.Lock()
 	defer v2Tracing.Unlock()
-	return v2Tracing.thriftClientMiddleware
+	if v2Tracing.thriftClientMiddlewareProvider != nil {
```

Comment:
you meant:
```suggestion
	if v2Tracing.thriftClientMiddlewareProvider == nil {
```
instead?